### PR TITLE
cre.cpp: fix isXPointerInCurrentPage()

### PR DESCRIPTION
**Pull request by poire-z**
Fix bookmark dogear not shown when revisiting some page when hanging punctuation is enabled and the first char of the page is hanging in the left margin. See https://github.com/koreader/koreader/pull/6456#issuecomment-993461335

Also add `cre.smoothScaleBlitBuffer()` to allow benchmarking BB scaling methods on devices.
See https://github.com/koreader/koreader/pull/3905#issuecomment-989789750

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader-base/1441)

### CODE REVIEW

**Comment by NiLuJe** [comment]
Possibly needs doc/a matching destructor, because it's not necessarily clear that the returned pointer is allocated and needs to be freed by the caller.

(That, or make it return an actual KO BlitBuffer flagged as allocated and stuff).

(I'll take a "don't use this" comment if this is just for testing purposes).

**Comment by poire-z** [comment]
I can add:
```c
/* This was added for testing and benchmarking purpose: do not use it.
 * Note that the returned pointer 'data' is allocated and needs to freed */
static int smoothScaleBlitBuffer(lua_State *L) {
```

Thought about having it returned a BB, but I don't know if that's possible with just blitbuffer.h included and having just this available:
https://github.com/koreader/koreader-base/blob/2e53592573718a59399df1739a92b39cf8d92e2d/blitbuffer.h#L124-L131
blitbuffer.lua .new() return this, or rather this possibly wrapped by metatype and Lua stuff that I'm not familiar with :)
Obviously, such stuff, if needed, shouldn't be done in the plain cre.cpp, but possibly by some other ffi wrapper, that I'd rather not add, or make cre.cpp dependant on some other stuff.

So, well, unless you see an obvious cheap way to return a classic usable Lua BB, I'll go with the comment :)

File: cre.cpp [hunk 1]
```diff
@@ -3545,6 +3549,19 @@ static int getImageDataFromPosition(lua_State *L) {
     return 0;
 }
 
+static int smoothScaleBlitBuffer(lua_State *L) {
```
Comment (Frenzie) [inline cre.cpp:3552]: Besides a comment you could also consider giving it a less safe name? (Just a thought; I'm fine with just the comment.)

**Comment by NiLuJe** [comment]
Comment's fine with me, yeah ;).

(I had a vague memory *something* might have been doing BBs-in-C-Lua, but, no, I was probably thinking of the C blitter, and that's FFI).

File: cre.cpp (hunk 1 above)
Comment (poire-z) [reply cre.cpp:3552]: (Well, with a less safe name, if we end up one day using it :) I'd have to make a PR here to rename it. With this safe pretty name, we can just use it and pretend I forgot to mark that comment as obsolete :)

**Pull request merged**